explainer.md

```diff
@@ -139,20 +176,66 @@ let depthValueInMeters = depthInfo.getDepth(x, y);
 
 ## Appendix: Proposed Web IDL
 
+The comments are left in place to provide additional context on the API to the readers of the IDL and should also be repeated in the explainer text above.
+
 ```webidl
+enum XRDepthSensingUsage {
+  "cpu-optimized",
+  "gpu-optimized"
+};
+
+enum XRDepthFormat {
+  // Has to be supported everywhere:
+  "luminance-alpha", // internal_format = LUMINANCE_ALPHA, type = UNSIGNED_BYTE,
+                     // 2 bytes per pixel, unpack via dot([R, A], [255.0, 256*255.0])
+  // Other formats do not have to be supported by the user agents:
+  "float32",         // internal_format = R32F, type = FLOAT
+};
+
+// Post-session-creation, the depth sensing state needs to be configured
+// with the desired data format.
+dictionary XRDepthSensingStateInit {
+  XRDepthSensingUsage usage = "cpu-optimized";
+  XRDepthFormat depthFormat = "luminance-alpha";
+};
+
+partial interface XRSession {
+  // Non-null iff depth-sensing is enabled:
+  readonly attribute XRDepthFormat? preferredDepthFormat;
+  readonly attribute XRDepthSensingUsage? preferredDepthSensingUsage;
+
+  // Can be called iff depth-sensing is enabled, will throw if unsupported format is passed in.
+  // The user agent can reject session creation if the desired usage is not supported.
+  // The user agent can support reconfiguring a session, but that is not required.
+  Promise<void> updateDepthSensingState(optional XRDepthSensingStateInit state = {});
```

Comment:
Not needed, but see the comment above (I don't think we can expose the preferred usage and format earlier than on a session).

Reply:
Other parts of WebXR usually queue tasks so a change in state happens between frames. Could it take several frames for this information to update?

Reply:
Yes, I think we won't be able to guarantee the timings here, so it may mean that you'll get several frames before you issued the request and the time it went in effect. What we should be able to guarantee is that all animation frames that you're getting after this promise has settled already have the desired setting in effect.

Reply:
OK, if we were to keep this, this would be hard to spec.

Reply:
Resolving, the latest version of the proposal no longer has this method.